crates/nu-command/src/filesystem/watch.rs

```diff
@@ -0,0 +1,295 @@
+                    Some(glob) => glob.matches_path(&path),
+                    None => true,
+                };
+                if verbose && glob_pattern.is_some() {
+                    eprintln!("Matches glob: {matches_glob}");
+                }
+
+                if matches_glob {
+                    let stack = &mut stack.clone();
+
+                    if let Some(position) = block.signature.get_positional(0) {
+                        if let Some(position_id) = &position.var_id {
+                            stack.add_var(
+                                *position_id,
+                                Value::String {
+                                    val: operation.to_string(),
+                                    span: call.span(),
+                                },
+                            );
+                        }
+                    }
+
+                    if let Some(position) = block.signature.get_positional(1) {
+                        if let Some(position_id) = &position.var_id {
+                            stack.add_var(
+                                *position_id,
+                                Value::String {
+                                    val: path.to_string_lossy().to_string(),
+                                    span: call.span(),
+                                },
+                            );
+                        }
+                    }
+
+                    if let Some(position) = block.signature.get_positional(2) {
+                        if let Some(position_id) = &position.var_id {
+                            stack.add_var(
+                                *position_id,
+                                Value::String {
+                                    val: new_path
+                                        .unwrap_or_else(|| "".into())
+                                        .to_string_lossy()
+                                        .to_string(),
+                                    span: call.span(),
+                                },
+                            );
+                        }
+                    }
+
+                    let eval_result = eval_block(
+                        engine_state,
+                        stack,
+                        &block,
+                        Value::Nothing { span: call.span() }.into_pipeline_data(),
+                        call.redirect_stdout,
+                        call.redirect_stderr,
+                    );
+
+                    match eval_result {
+                        Ok(val) => {
+                            val.print(engine_state, stack)?;
+                        }
+                        Err(err) => {
+                            // TODO: this isn't as nice as the Miette errors. Find a way to print those.
+                            // Unfortunately can't just wrap err in PipelineData, PipelineData.print() doesn't work that way
+                            eprintln!("{err:?}");
```

Comment:
It looks like there's ways to call miette without doing it how we normally do it. I'm guessing you could experiment with something like this.
https://github.com/zkat/miette/blob/2451ad6a963c222831977e89542a7349b66f11cf/tests/graphical.rs#L9

Reply:
Hmm, that may be a bit tricky. `GraphicalReportHandler` has a `render_report()` fn that might work, but:
1. Setting up the `GraphicalReportHandler` properly requires knowledge of some terminal-related info (like: current width, unicode capability) and that feels like it's getting into the `nu-cli` crate's territory
2. `render_report()`'s documentation suggests that I probably shouldn't use it like this:
> /// Render a [`Diagnostic`]. This function is mostly internal and meant to
> /// be called by the toplevel [`ReportHandler`] handler, but is made public
>  /// to make it easier (possible) to test in isolation from global state.

I think I need to 1) understand Miette a little better 2) think about whether there's a clean way we can pass these errors off to something in `nu-cli` to be printed. Will do that, but in the meantime I think the current error printing is good enough to experiment with `watch` and see if we like it.

Reply:
you might ask kat in discord in the external-collab channel.

Reply:
do we want to wait on the improvements, or go ahead and land now and improve as we learn more?

Reply:
Personally I'd like to land it now. I'd like to get this into people's hands so I can solicit feedback (and I'll follow up on adding nice Miette diagnostics when the block fails).